torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -135,6 +143,30 @@ void ${kernelName}(IndexType totalElements, ${formals}) {
 }
 )");
 
+auto cpu_compilation_unit_template = CodeTemplate(R"(
+#include <cstddef>
+#include <math.h>
+#include <iostream>
+${type_declarations}
+
+static void ${kernelName}_kernel(IndexType totalElements, ${formals}) {
+  // TODO: parallelize with something reasonable
+  for (IndexType linearIndex = 0;
+        linearIndex < totalElements;
+        linearIndex += 1) {
+      // Convert `linearIndex` into an offset of tensor:
+      ${tensorOffsets}
+      // calculate the results
+      ${kernelBody}
+    }
+}
+
+extern "C"
```

Comment:
I wonder if it's more portable to generate C code and call a C compiler on it (e.g., you don't  have to worry about linking against libstdc++). Are we using/plan to use C++ features in a nontrivial way?

Reply:
We already use a template for the TensorInfo struct, so my gut is that we will.